### PR TITLE
gnu-getopt: check `getopt -T`

### PR DESCRIPTION
**Pull request by LecrisUT**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I believe `getopt -o --test` does the same as `getopt -T` and checking the return code is `4`, but it's worth adding this test as a sanity check. I had a script that was using this, and it seems to have failed the test.

### CODE REVIEW

**Comment by chenrui333** [comment]
@LecrisUT thanks for your PR, can you amend your commit msg as "getopt: check `getopt -T`"?

**Comment by LecrisUT** [comment]
For reference, [this](https://github.com/LecrisUT/homebrew-tmt/actions/runs/8144364622) is the run that I was working that prompted this check. It seems in some recipes we have to replace `getopt` with fully-qualified path?

File: Formula/g/gnu-getopt.rb [hunk 1]
```diff
@@ -41,5 +41,8 @@ def install
 
   test do
     system "#{bin}/getopt", "-o", "--test"
+    # Check that getopt is enhanced
+    system "#{bin}/getopt", "-T"
+    assert_equal 4, $CHILD_STATUS.exitstatus
```
Comment (LecrisUT) [inline Formula/g/gnu-getopt.rb:46]: Welp, that didn't work. What should be used here instead of `system`?

**Comment by chenrui333** [comment]
sorry, the commit msg should be actually starting with `gnu-getopt` (which is the formula name)

File: Formula/g/gnu-getopt.rb [hunk 2]
```diff
@@ -41,5 +41,8 @@ def install
 
   test do
     system "#{bin}/getopt", "-o", "--test"
+    # Check that getopt is enhanced
+    quiet_system "#{bin}/getopt", "-T"
+    assert_equal 4, $CHILD_STATUS.exitstatus
```
[the remark this answers is not in the document]
Comment (LecrisUT) [reply Formula/g/gnu-getopt.rb:46]: Tests for enhanced version

---

https://linux.die.net/man/1/getopt

> -T, --test
    Test if your getopt(1) is this enhanced version or an old version. This generates no output, and sets the error status to 4. Other implementations of getopt(1), and this version if the environment variable GETOPT_COMPATIBLE is set, will return '--' and error status 0.

[the remark this answers is not in the document]
Comment (chenrui333) [reply Formula/g/gnu-getopt.rb:46]: 👍

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks @LecrisUT!

**Pull request merged**